src/element.ts

```diff
@@ -491,4 +488,95 @@ export abstract class Element {
     const fontInfo = this.fontInfo;
     this.setFont(fontInfo.family, fontInfo.size, weight, fontInfo.style);
   }
+
+  /** Get element width. */
+  getWidth(): number {
+    return this.width;
+  }
+
+  /** Set element width. */
+  setWidth(width: number): this {
+    this.width = width;
+    return this;
+  }
+
+  /** Shift element down `yShift` pixels. Negative values shift up. */
+  setYShift(yShift: number): this {
+    this.yShift = yShift;
+    return this;
+  }
+
+  /** Get shift element `yShift` */
+  getYShift(): number {
+    return this.yShift;
+  }
+
+  /**
+   * Set shift element right `xShift` pixels. Negative values shift left.
+   */
+  setXShift(xShift: number): this {
+    this.xShift = xShift;
+    return this;
+  }
+
+  /** Get shift element `xShift` */
+  getXShift(): number {
+    return this.xShift;
+  }
+
+  /**
+   * Set element text
+   */
+  setText(text: string): this {
+    this.text = text;
+    return this;
+  }
+
+  /** Get element text */
+  getText(): string {
+    return this.text;
+  }
+
+  /**
+   * Render the element text .
+   */
+  renderText(ctx: RenderContext, xPos: number, yPos: number): void {
+    ctx.save();
+    ctx.setFont(this.textFont);
+    ctx.fillText(this.text, xPos + this.xShift, yPos + this.yShift);
+    ctx.restore();
+  }
+
+  /** Canvas used to measure text */
+  static #txtCanvas?: HTMLCanvasElement;
+
+  measureText(): TextMetrics {
+    let txtCanvas = Element.#txtCanvas;
+    if (!txtCanvas) {
+      // Create the SVG text element that will be used to measure text in the event
+      // of a cache miss.
+      txtCanvas = document.createElement('canvas');
+      Element.#txtCanvas = txtCanvas;
+    }
+    const context = txtCanvas.getContext('2d');
+    if (!context) throw new RuntimeError('Font', 'No txt context');
+    context.font = Font.toCSSString(Font.validate(this.textFont));
+    this.textMetrics = context.measureText(this.text);
+    this.boundingBox = new BoundingBox(
+      0,
+      -this.textMetrics.actualBoundingBoxAscent,
+      this.textMetrics.width,
+      this.textMetrics.actualBoundingBoxDescent + this.textMetrics.actualBoundingBoxAscent
```

Comment:
We could extract `this.textMetrics.actualBoundingBoxAscent` => `const ascent = this.textMetrics.actualBoundingBoxAscent`

Reply:
done, height also modified to be a member variable.